common/src/dodeca.rs

```diff
@@ -255,97 +260,133 @@ lazy_static! {
+                        Side::from_index(a),
+                        Side::from_index(b),
+                        Side::from_index(c),
+                    ];
                     vertex += 1;
                 }
             }
         }
         assert_eq!(vertex, 20);
         result
-    };
+    })
+}
 
-    // Which vertices are adjacent to other vertices and opposite the canonical sides
-    static ref ADJACENT_VERTICES: [[Vertex; 3]; VERTEX_COUNT] = {
+// Which vertices are adjacent to other vertices and opposite the canonical sides
+fn adjacent_vertices_static() -> &'static [[Vertex; 3]; VERTEX_COUNT] {
+    static LOCK: OnceLock<[[Vertex; 3]; VERTEX_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| {
         let mut result = [[Vertex::A; 3]; VERTEX_COUNT];
 
-        for vertex in 0..VERTEX_COUNT {
+        for (i, triple) in result.iter_mut().enumerate().take(VERTEX_COUNT) {
             for result_index in 0..3 {
-                let mut test_sides = VERTEX_SIDES[vertex];
+                let mut test_sides = vertex_sides_static()[i];
                 // Keep modifying the result_index'th element of test_sides until its three elements are all
                 // adjacent to a single vertex. That vertex is the vertex we're looking for.
                 for side in Side::iter() {
-                    if side == VERTEX_SIDES[vertex][result_index] {
+                    if side == vertex_sides_static()[i][result_index] {
                         continue;
                     }
                     test_sides[result_index] = side;
-                    if let Some(adjacent_vertex) = Vertex::from_sides(test_sides[0], test_sides[1], test_sides[2]) {
-                        result[vertex][result_index] = adjacent_vertex;
+                    if let Some(adjacent_vertex) =
+                        Vertex::from_sides(test_sides[0], test_sides[1], test_sides[2])
+                    {
+                        triple[result_index] = adjacent_vertex;
                     }
                 }
             }
         }
         result
-    };
+    })
+}
 
-    /// Transform that converts from cube-centric coordinates to dodeca-centric coordinates
-    static ref DUAL_TO_NODE_F64: [na::Matrix4<f64>; VERTEX_COUNT] = {
-        let mip_origin_normal = math::mip(&math::origin(), &SIDE_NORMALS_F64[0]); // This value is the same for every side
+/// Transform that converts from cube-centric coordinates to dodeca-centric coordinates
+fn dual_to_node_f64_static() -> &'static [na::Matrix4<f64>; VERTEX_COUNT] {
+    static LOCK: OnceLock<[na::Matrix4<f64>; VERTEX_COUNT]> = OnceLock::new();
+    LOCK.get_or_init(|| {
+        let mip_origin_normal = math::mip(&math::origin(), &side_normals_f64_static()[0]); // This value is the same for every side
         let mut result = [na::zero(); VERTEX_COUNT];
-        for i in 0..VERTEX_COUNT {
-            let [a, b, c] = VERTEX_SIDES[i];
+        for (i, item) in result.iter_mut().enumerate().take(VERTEX_COUNT) {
```

Comment:
`take` is redundant here: `result.iter_mut()` already has a length of exactly `VERTEX_COUNT`.

Also, please keep unrelated refactoring in a separate commit. This makes it much easier to ensure that each change has the intended effect when reviewing.